### PR TITLE
.github: add pull request template and contributing guide

### PR DESCRIPTION
**Pull request by zz-jason**
This pull request adds two files for:

- pull request template, to help contributors summarize their contributions.
- contributing guide, to help contributors getting started with how to contribute to this repo.

### CODE REVIEW

File: .github/pull_request_template.md [hunk 1]
```diff
@@ -0,0 +1,13 @@
+<!-- Thank you for contributing to TiDB Development Guide!  -->
+<!-- Please follow the PR title format:                     -->
+<!--    "section: what's changed"                           -->
+
```
Comment (tisonkun) [inline .github/pull_request_template.md:4]:
```suggestion

<!-- This repo turns on "Require signed commits" option. See also [this page](https://docs.github.com/en/github/authenticating-to-github/managing-commit-signature-verification/about-commit-signature-verification) to understand how to satisfy the requirement. -->

```

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+2. Improve the guide content, for example:
+    - fix typos and adjust some statements to make them clearer
+    - modify the contents if it's outdated
+    - add new pages if something is missing
+    - adjust the contents if the article structure is unreasonable
+
+3. Reviewing the pull request
+
+Once the pull request receives 2 LGTM from the maintainers, it can be merged.
+
```
Comment (tisonkun) [inline CONTRIBUTING.md:24]:
```suggestion

**Note**: This repo turns on "Require signed commits" option. See also [this page](https://docs.github.com/en/github/authenticating-to-github/managing-commit-signature-verification/about-commit-signature-verification) to understand how to satisfy the requirement.

```

**Review by tisonkun — CHANGES_REQUESTED** [review]
I'd prefer highlight we turn on "Require signed commits" since not every GitHub user aware of it.

**Review by tisonkun — APPROVED** [review]

**Comment by zz-jason** [comment]
@all-contributors please add @zz-jason for content

**Pull request merged**